app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -40,6 +40,17 @@ def self.default_blacklisted_event_names
     )
   end
 
+  def hostname_uniqueness_valid?
+    return unless hostname_required?
+    return unless hostname.present? # Presence is checked elsewhere
+
+    existing_providers = Endpoint.where(:hostname => hostname.downcase)
+                                 .where.not(:resource_id => id).includes(:resource)
+                                 .select { |endpoint| endpoint.resource.uid_ems == keystone_v3_domain_id }
```

Comment:
Do we have to worry about the case where keystone v2 is selected?  No, because the domain_id will be nil in that case?